src/goto-instrument/cover_basic_blocks.cpp

```diff
@@ -153,12 +153,17 @@ void cover_basic_blockst::add_block_lines(
   cover_basic_blockst::block_infot &block,
   const goto_programt::instructiont &instruction)
 {
-  const irep_idt &line = instruction.source_location.get_line();
-  if(!line.empty())
-  {
-    block.lines.insert(unsafe_string2unsigned(id2string(line)));
-    block.source_lines.insert(instruction.source_location);
-  }
+  const auto &add_location = [&](const source_locationt &location) {
+    const irep_idt &line = location.get_line();
+    if(!line.empty())
+    {
+      block.lines.insert(unsafe_string2unsigned(id2string(line)));
```

Comment:
:question: `block_infot::lines` is a set I presume?

Reply:
Yes.